src/java.base/unix/native/libnio/ch/FileChannelImpl.c

```diff
@@ -175,8 +175,39 @@ Java_sun_nio_ch_FileChannelImpl_transferTo0(JNIEnv *env, jobject this,
     jint dstFD = fdval(env, dstFDO);
 
 #if defined(__linux__)
+    // Check whether the destination is in append mode: copy_file_range fails
+    // with EBADF in this case and sendfile fails with EINVAL.
+    int dstFlags = fcntl(dstFD, F_GETFL);
+    if (dstFlags < 0) {
+        JNU_ThrowIOExceptionWithLastError(env, "Destination flags");
+        return IOS_THROWN;
+    }
+    if ((dstFlags & O_APPEND) != 0)
+        return IOS_UNSUPPORTED_CASE;
+
     off64_t offset = (off64_t)position;
-    jlong n = sendfile64(dstFD, srcFD, &offset, (size_t)count);
+    jlong n;
+    if (my_copy_file_range_func) {
```

Comment:
I assume this should check if my_copy_file_range_func is NULL.

Reply:
Yes, it should.